deploy/routeros.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+#Here is a script to deploy cert to routeros router.
+
+#returns 0 means success, otherwise error.
+
+########  Public functions #####################
+
+#domain keyfile certfile cafile fullchain
+routeros_deploy() {
+  _cdomain="$1"
+  _ckey="$2"
+  _ccert="$3"
+  _cca="$4"
+  _cfullchain="$5"
+
+  _debug _cdomain "$_cdomain"
+  _debug _ckey "$_ckey"
+  _debug _ccert "$_ccert"
+  _debug _cca "$_cca"
+  _debug _cfullchain "$_cfullchain"
+
+  if [ -z "$ROUTER_OS_HOST" ]; then
+    _debug "Using _cdomain as ROUTER_OS_HOST, please set if not correct."
+    ROUTER_OS_HOST="$_cdomain"
+  fi
+
+  if [ -z "$ROUTER_OS_USERNAME" ]; then
+    _err "Need to set the env variable ROUTER_OS_USERNAME"
+    return 1
+  fi
+
+  _info "Trying to push key '$_ckey' to router"
+  scp "$_ckey" "$ROUTER_OS_USERNAME@$ROUTER_OS_HOST:$_cdomain.key"
+  _info "Trying to push cert '$_ccert' to router"
+  scp "$_ccert" "$ROUTER_OS_USERNAME@$ROUTER_OS_HOST:$_cdomain.cer"
+  _info "Trying to push ca cert '$_cca' to router"
+  scp "$_cca" "$ROUTER_OS_USERNAME@$ROUTER_OS_HOST:$_cdomain.ca"
```

Comment:
I guess that is true as long as the client accessing the router has access to internet, if using a hotspot with https this might not be the case. Can still remove this from the script

Reply:
I just noticed ... what I'm doing on my Mikrotiks is uploading `$_cfullchain.cer` as the server key. This file includes the CA chain, and because it's in the same file, it makes the web server include the CA chain when presenting its certificate to the client. The `www-ssl` server doesn't have a way to configure more than one cert, so I *think* if you upload the files separately, the server will only send the certificate from `server.cer`, without the rest of the CA chain, which could cause problems for some clients.

Reply:
will modify to use the _cfullchain instead